src/python/pants/backend/terraform/dependency_inference.py

```diff
@@ -41,7 +41,8 @@ class TerraformHcl2Parser(PythonToolRequirementsBase):
     options_scope = "terraform-hcl2-parser"
     help = "Used to parse Terraform modules to infer their dependencies."
 
-    default_version = "python-hcl2==3.0.5"
+    default_version = "python-hcl2==4.3.0"
+    default_requirements = ["python-hcl2>=3.0.5,<5"]
```

Comment:
```suggestion
    default_requirements = ["python-hcl2>=3.0.5,<5"]
    default_version = default_requirements[0]
```

Reply:
As mentioned, I'd prefer not to, and in this case the default_requirements and the default_version are different by design (we know we're compatible with a wider range, but we want to generate a lockfile for this specific version)